### PR TITLE
[IMP] web: throw error when a domain is invalid

### PR DESCRIPTION
**Pull request by FrancoisGe**

### CODE REVIEW

File: addons/web/static/src/core/domain.js [hunk 1]
```diff
@@ -151,19 +151,25 @@ function normalizeDomainAST(domain, op = "&") {
...
   }
   let values = domain.value.slice();
-  while (expected > 0) {
-    expected--;
+  while (expected < 0) {
+    expected++;
     values.unshift({ type: 1 /* String */, value: op });
   }
+  if (expected > 0) {
+    throw new Error(`invalid domain ${formatAST(domain)} (missing ${expected} segment(s))`);
```
Comment (ged-odoo) [inline addons/web/static/src/core/domain.js]: probably should create a `InvalidDomainError` class, subclassing OdooError.

**Comment by Xavier-Do** [comment]
just testing stuff, ignore me. 

@robodoo r+

**Comment by Xavier-Do** [comment]
@robodoo r+

**Comment by Xavier-Do** [comment]
sudo
@robodoo r+

**Comment by Xavier-Do** [comment]
@robodoo r-

**Comment by ged-odoo** [comment]
robodoo r+

**Pull request closed without merging**